File: map/cheltu/src/Domain/Cheltuiala.java
package Domain;

import java.util.Date;

public class Cheltuiala implements HasID<String> {

    private String idChelt;
    private TipCheltuiala tip;
    private double suma;
    private String desc;
    private String efectuatDe;
    private Date data;

    public Cheltuiala(TipCheltuiala tip, double suma, String desc, String efectuatDe, Date data) {
        this.tip = tip;
        this.suma = suma;
        this.desc = desc;
        this.efectuatDe = efectuatDe;
        this.data = data;
        this.idChelt=efectuatDe+data.toString()+suma;
    }

    @Override
    public String getId() {
        return idChelt;
    }

    @Override
    public void setId(String integer) {
        idChelt=integer;
    }


    public TipCheltuiala getTip() {
        return tip;
    }

    public void setTip(TipCheltuiala tip) {
        this.tip = tip;
    }

    public double getSuma() {
        return suma;
    }

    public void setSuma(double suma) {
        this.suma = suma;
    }

    public String getDesc() {
        return desc;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }

    public String getEfectuatDe() {
        return efectuatDe;
    }

    public void setEfectuatDe(String efectuatDe) {
        this.efectuatDe = efectuatDe;
    }

    public Date getData() {
        return data;
    }

    public void setData(Date data) {
        this.data = data;
    }

    @Override
    public String toString() {
        return tip+";"+suma+";"+desc+";"+efectuatDe+";"+data+";";
    }
}

File: map/cheltu/cheltu/Cheltuiala.cs
using System;

namespace cheltu
{
    public class Cheltuiala
    {
       
        public string id { get; set; }
        public string desc { get; set; }
        public string efectuatDe { get; set; }
        public TipCheltu tip { get; set; }
        public double suma { get; set; }
        public DateTime data { get; set; }

        public Cheltuiala(string desc, string efectuatDe, TipCheltu tip, double suma, DateTime data)
        {
            this.desc = desc;
            this.efectuatDe = efectuatDe;
            this.tip = tip;
            this.suma = suma;
            this.data = data;
            this.id = efectuatDe + data;
        }

        public override string ToString()
        {
            return suma + efectuatDe + data + tip + desc;
        }
    }
}